Stratis.Bitcoin/Features/RPC/Controllers/ConnectionManagerController.cs

```diff
@@ -35,5 +39,60 @@ public bool AddNode(string endpointStr, string command)
             }
             return true;
         }
+
+        /// <summary>
+        /// RPC implementation of getpeerinfo
+        /// </summary>
+        /// <returns>List of RPC peer nodes</returns>
+        [ActionName("getpeerinfo")]
+        public List<PeerNodeModel> GetPeerInfo()
+        {
+            List<PeerNodeModel> peerList = new List<PeerNodeModel>();
+            int index = 0;
+            foreach (Node node in this.ConnectionManager.ConnectedNodes)
```

Comment:
I wonder if it would be a good idea to create an iteration copy of ConnectedNodes.